File: app/src/main/java/io/mgba/Presenter/Interfaces/IMainPresenter.java
package io.mgba.Presenter.Interfaces;

import android.content.Intent;
import android.view.MenuItem;
import io.mgba.Model.Interfaces.ILibrary;

public interface IMainPresenter {

    void onActivityResult(int requestCode, int resultCode, Intent intent);
    void onDestroy();
    ILibrary getILibrary();
    void onMenuItemSelected(MenuItem item);
    void onSearchTextChanged(String oldQuery, String newQuery);

}

File: app/src/main/java/io/mgba/presenter/interfaces/IMainPresenter.kt
package io.mgba.presenter.interfaces

import android.content.Intent
import android.view.MenuItem
import io.mgba.model.interfaces.ILibrary

interface IMainPresenter {
    val iLibrary: ILibrary?

    fun onActivityResult(requestCode: Int, resultCode: Int, intent: Intent)
    fun onDestroy()
    fun onMenuItemSelected(item: MenuItem)
    fun onSearchTextChanged(oldQuery: String, newQuery: String)

}
